src/lib/constants.js

```diff
@@ -46,3 +46,241 @@ export const RETURN_KEY = 'Enter';
 export const ESCAPE_KEY = 'Escape';
 export const COMMAND_KEY = 'Meta';
 export const CONTROL_KEY = 'Control';
+
+/**
+ * Trace type constants
+ */
+export const CHART_CLASS = {
+  POINTS_2D: 'POINTS_2D',
+  TERNARY: 'TERNARY',
+  PLOT_3D: 'PLOT_3D',
+  GEOGRAPHIC: 'GEOGRAPHIC',
+  MAPBOXGL: 'MAPBOXGL',
+  SINGLE: 'SINGLE',
+  FINANCIAL: 'FINANCIAL',
+  TABLE: 'TABLE',
+};
+
+// Layout specification for TraceTypeSelector.js
+export const CHART_CATEGORY = {
+  BUSINESS: 'BUSINESS',
+  SCIENCE: 'SCIENCE',
+  CHARTS_3D: 'CHARTS_3D',
+  FINANCIAL: 'FINANCIAL',
+  STATISTICS: 'STATISTICS',
+  MAPS: 'MAPS',
+};
+
+export const CATEGORY_LAYOUT = [
+  {category: CHART_CATEGORY.BUSINESS, label: 'Business'},
+  {category: CHART_CATEGORY.SCIENCE, label: 'Science'},
+  {category: CHART_CATEGORY.CHARTS_3D, label: '3D charts'},
+  {category: CHART_CATEGORY.FINANCIAL, label: 'Finance'},
+  {category: CHART_CATEGORY.STATISTICS, label: 'Statistics'},
+  {category: CHART_CATEGORY.MAPS, label: 'Maps'},
+];
+
+export const TRACE_TYPES = {
+  scatter: {
+    meta: {
```

Comment:
Much of these consts were pulled from `streambed` I was not sure how much of it should stay / how much of it should go. 

What do you think about moving https://github.com/plotly/react-plotly.js-editor/blob/master/src/components/fields/TraceSelector.js#L21 into the `constants.js` file?

Reply:
We don't have to use what streambed has : ) we just need the logic (basically which chart types go into which categories). Plus our new requirement now is that users can make that configurable. So the best data structure is the one that's simplest to use in an API.

About https://github.com/plotly/react-plotly.js-editor/blob/master/src/components/fields/TraceSelector.js#L21 I think they're here because trace names need localization, and that localize function must wrap a component to make the localize prop available to it.

We should probably be making localize available at the very top level of our app, localizing component per component is a bit rough and unecesary..

I don't mind moving all trace types to the constants folder though..

Reply:
there should definitely be one canonical mapping from trace type to localized trace name, so that if I have access to `"scatter"` and `_` I should be able to get at the localized name. Makes sense to move that into `constants` and pack in the default categorization as well IMO.